bin/mympirun.py

```diff
@@ -42,81 +42,138 @@
 
 import sys
 import os
+import traceback
+import glob
+import inspect
 
 from vsc.utils import fancylogger
-from vsc.mympirun.mpi.factory import getinstance
-from vsc.mympirun.mpi.mpi import whatMPI
+from vsc.mympirun.rm.factory import getinstance
+import vsc.mympirun.mpi.mpi
 from vsc.mympirun.option import MympirunOption
-from vsc.mympirun.rm.sched import whatSched
+import vsc.mympirun.rm.sched
 
 _logger = fancylogger.getLogger()
+fancylogger.setLogLevelInfo()
 
 
-class ExitException(Exception):
-    """Exception thrown when we wish to exit, but no real errors occured"""
-    pass
+def get_mpi_and_sched_and_options():
+    """Parses the mpi and scheduler based on current environment and
+    guesses the best one to use
 
+    returns:
+    mpi             --
+    sched           --
+    mo              --
+    """
+    _logger.debug("mympirun.py - get_mpi_and_sched_and_options()")
 
-def get_mpi_and_sched_and_options():
-    """Parses the mpi and scheduler based on current environment and guesses the best one to use"""
-    scriptname, mpi, found_mpi = whatMPI(sys.argv[0])
+    import_implementations(vsc.mympirun.mpi.mpi)
+    import_implementations(vsc.mympirun.rm.sched)
+
+    setmpi_index = sys.argv.index("-M") if "-M" in sys.argv else 0
+    setmpi = sys.argv[setmpi_index+1] if setmpi_index else sys.argv[0]
+    _logger.debug("mympirun.py - setmpi: %s" % setmpi)
+
+    scriptname, mpi, found_mpi = vsc.mympirun.mpi.mpi.whatMPI(setmpi)
+    _logger.debug("mympirun.py - whatMPI returned scriptname: %s, mpi: %s, found_mpi: %s" %
+                 (scriptname, mpi, found_mpi))
 
     ismpirun = scriptname == 'mpirun'
 
     mo = MympirunOption(ismpirun=ismpirun)
```

Comment:
`MympirunOption(ismpirun=ismpirun)` depends on `ismpirun`, which depends on `whatMPI()`, which depends on `setmpi`. I haven't thoroughly checked if the `ismpirun=ismpirun` is absolutely necessary but the original script had this and it affects some stuff later in the script, so I kept it.